src/integritysetup/integritysetup-generator.c

```diff
@@ -0,0 +1,224 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "alloc-util.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "fstab-util.h"
+#include "generator.h"
+#include "hexdecoct.h"
+#include "id128-util.h"
+#include "main-func.h"
+#include "mkdir.h"
+#include "parse-util.h"
+#include "path-util.h"
+#include "proc-cmdline.h"
+#include "specifier.h"
+#include "string-util.h"
+#include "unit-name.h"
+
+static const char *arg_dest = NULL;
+static const char *arg_integritytab = NULL;
+static const char *arg_integritysetupbin = NULL;
+static char *arg_options = NULL;
+STATIC_DESTRUCTOR_REGISTER(arg_options, freep);
+
+static void option(
+                FILE *o,
+                char *token) {
+
+        char *save_ptr = NULL;
+        char *match = strtok_r(token, "=", &save_ptr);
+        if (match) {
+                if (strlen(match) > 1)
+                        fprintf(o, "-");
+
+                fprintf(o, "-%s ", match);
+                match = strtok_r(NULL, "=", &save_ptr);
+                if (match)
+                        fprintf(o, "%s ", match);
+        }
+}
+
+static int options_string(
+                char *options,
+                char **options_formatted) {
+
+        if (options && options_formatted  && *options_formatted == NULL && strlen(options) > 0) {
```

Comment:
we prefer to exit early to reduce indentation, ie return einval immediately if the reversed check is true

Reply:
Done